server/container/guice/mailet/src/test/java/org/apache/james/utils/GuiceMailetLoaderTest.java

```diff
@@ -183,4 +184,25 @@ public void getMailetShouldAllowCustomInjections() throws Exception {
 
         assertThatCode(() -> mailet.service(FakeMail.defaultFakeMail())).doesNotThrowAnyException();
     }
+
+    @Ignore("JAMES-2866 singleton are not shared between extensions")
+    @Test
+    public void allMailetsShouldShareTheSameSingleton() throws Exception {
+        GuiceGenericLoader genericLoader = new GuiceGenericLoader(
+            Guice.createInjector(),
+            new ExtendedClassLoader(RECURSIVE_CLASSPATH_FILE_SYSTEM),
+            new ExtensionConfiguration(ImmutableList.of(new ClassName("org.apache.james.transport.mailets.MyExtensionModule"))));
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(genericLoader, NO_MAILET_CONFIG_OVERRIDES);
+
+        Mailet mailet1 = guiceMailetLoader.getMailet(FakeMailetConfig.builder()
+            .mailetName("MyGenericMailet")
+            .mailetContext(FakeMailContext.defaultContext())
+            .build());
+        Mailet mailet2 = guiceMailetLoader.getMailet(FakeMailetConfig.builder()
+            .mailetName("MyGenericMailet")
+            .mailetContext(FakeMailContext.defaultContext())
+            .build());
+
+        assertThat(mailet1).isEqualTo(mailet2);
```

Comment:
you should make the assertion more obvious:
```
assertThat(mailet1.dependency).isSameAs(mailet2.dependency);
```

Reply:
Sure, I thought about it, but then I need to know that it is a MyCustomMailet, and also to have the dependency on the classpath to be able to call non API method.

Here I prefer adding a description.

Reply:
Ok, I understand, but using equals for that purpose sounds really brittle